manticore/platforms/linux.py

```diff
@@ -1331,7 +1337,57 @@ def sys_rt_sigprocmask(self, cpu, how, newset, oldset):
     def sys_sigprocmask(self, cpu, how, newset, oldset):
         logger.debug("SIGACTION, Ignoring changing signal mask set cmd:%d", how)
         return 0
+    
+    def sys_dup(self, fd):
+        '''
+        Duplicates an open file descriptor
+        :rtype: int
+        :param fd: the open file descriptor to duplicate.
+        :return: the new file descriptor.
+        '''
+    
+        if not self._is_fd_open(fd):
+            logger.info("DUP: Passed fd is not open. Returning EBADF")
+            return -errno.EBADF
+        
+        newfd = self._dup(fd)
+        logger.debug('sys_dup(%d) -> %d', fd, newfd)
+        return newfd
+        
+    def sys_dup2(self, fd, newfd):
+        '''
+        Duplicates an open fd to newfd. If newfd is open, it is first closed
+        :rtype: int
+        :param fd: the open file descriptor to duplicate.
+        :param newfd: the file descriptor to alias the file described by fd.
+        :return: newfd.
+        '''
+        #TODO(yan): Once getrlimit(2) is implemented for getdtablesize, move
+        # this to a platform instance var. 256 is a conservative size that gets
+        # returned if rlimit() errors.
+        MAX_DTABLE_SIZE = 256
```

Comment:
can we move this to an instance variable now? and move this comment to a stub getrlimit(2) that just raises NotImplementError? also out of curiosity where is the dtable terminology used? i don't see it in the getrlimit man page or kernel source around getrlimit.

Reply:
Check man pages for getdtablesize(3) and sure.